### PR TITLE
Metadata module

### PR DESCRIPTION
**Pull request by nikmd23**
A not-yet-complete list of _requirements_ for the metadata system:
- Serve metadata from a resource (#23)
- Tell clients which filtering mechanisms are available
- Tell clients about URI templates to various resources
- Provide cache busting hash
- Provide list of SPA Client extension URLs?
- Any agent specific metadata?

This is a list of _requirements_, not tasks. As such, this list is not work to be completed, but rather things to be considered when designing the metadata module.

### CODE REVIEW

**Comment by nikmd23** [comment]
@avanderhoorn, this is getting close. Might be time for you to take a look at it too.

File: src/Glimpse.Client.Web/SpaClientResource.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+            await response.WriteAsync(@"
+<html>
+<head><meta http-equiv='refresh' content='3; url=/glimpse/client/index.html' ></head>
+<body><h1>We must reconcile this with <a href='/glimpse/client/index.html'>ClientResource</a>.</h1><p>Auto redirect in 3 seconds....</p></body>
+</html>");
+        }
+
+        public string Name => "SpaClientScript";
+        public ResourceParameters Parameters => new ResourceParameters(+ResourceParameter.Hash, ResourceParameter.RequestId);
```
Comment (khellang) [inline src/Glimpse.Client.Web/SpaClientResource.cs:25]: What is this `+` doing there? :smile:

Comment (khellang) [inline src/Glimpse.Client.Web/SpaClientResource.cs:25]: Ah, nevermind. Was too lazy to check. https://github.com/Glimpse/Glimpse.Prototype/blob/fb18c4605a14f72814e3c4abb5faf71d9b4d49c5/src/Glimpse.Server.Web/Framework/ResourceParameter.cs#L41-L44

Comment (nikmd23) [inline src/Glimpse.Client.Web/SpaClientResource.cs:25]: Yeah, I know - it's a little unconventional, but we're trying to keep that terse. 

Like? Dislike?

Comment (khellang) [inline src/Glimpse.Client.Web/SpaClientResource.cs:25]: Sure, it's neat, but it's wasn't exactly intuitive. Does it even show up in IntelliSense?

Why does it have to be that terse? I think I'd go for something like `ResourceParameter.Hash(isRequired: true)`.

**Pull request merged**